src/sbd-inquisitor.c

```diff
@@ -39,6 +39,28 @@ bool sync_resource_startup = false;
 
 int parse_device_line(const char *line);
 
+static int
+sanitize_numeric_option_value(const char *value)
+{
+    char *end = NULL;
+    int result = -1;
+
+    if (value == NULL) {
+        return -1;
+    }
+
+    errno = 0;
+
+    result = strtol(value, &end, 10);
+    if (result == LONG_MIN || result == LONG_MAX || errno != 0) {
```

Comment:
clang complains with good reason that we are comparing 'int' to the 'long int' constants.
Comparing directly to INT_MAX & INT_MIN won't work either as the cast isn't satturating.
But I guess when having result as 'long int' we could catch all cases with 'result <= INT_MIN'
and 'result >= INT_MAX' as 'LONG_MIN <= INT_MIN' and 'LONG_MAX >= INT_MAX'.
Cast would then happen at return when we've already assured range.
Probably better than going with a 'long int' return as all our use-cases are assigning to an 'int'
as far as I've seen.

Reply:
Yes I was expecting gcc to complain about that but.. even with -Wall or -Wpedantic flag he doesn't give any warning.  
`result <= INT_MIN || result >= INT_MAX` is the right way.

Reply:
Guess for that to work everywhere we need result to be 'long int'.

Reply:
Yep, well then I will change the return value to `long int` and will stick to the LONG_MIN and LONG_MAX constrains.

Reply:
But then we'd need to check for the range on use as everything we are assigning the return value to is 'int'.
Guess least effort is having result 'long int', check for INT-range and do cast on return.